modules/auxiliary/admin/sap/sap_igs_xmlchart_xxe.rb

```diff
@@ -0,0 +1,346 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+        update_info(
+          info,
+          'Name' => 'SAP Internet Graphics Server (IGS) XMLCHART XXE',
+          'Description' => %q{
+            This module exploits CVE-2018-2392 and CVE-2018-2393, two XXE vulnerabilities within the XMLCHART page
+            of SAP Internet Graphics Servers (IGS) running versions 7.20, 7.20EXT, 7.45, 7.49, or 7.53. Successful
+            exploitation will allow unauthenticated remote attackers to read files from the server as the user from which
+            the IGS service is started, which will typically be the SAP admin user. Alternatively attackers may also
+            be able to abuse the XXE to conduct a denial of service attack against the vulnerable SAP IGS server.
+          },
+          'Author' => [
+            'Yvan Genuer', # @_1ggy The researcher who originally found this vulnerability
+            'Vladimir Ivanov' # @_generic_human_ This Metasploit module
+          ],
+          'License' => MSF_LICENSE,
+          'References' => [
+            [ 'CVE', '2018-2392' ],
+            [ 'CVE', '2018-2393' ],
+            [ 'URL', 'https://download.ernw-insight.de/troopers/tr18/slides/TR18_SAP_IGS-The-vulnerable-forgotten-component.pdf' ]
+          ],
+          'Actions' => [
+            [ 'READ', { 'Description' => 'Remote file read' } ],
+            [ 'DOS', { 'Description' => 'Denial Of Service' } ]
+          ],
+          'DefaultAction' => 'READ',
+          'DefaultOptions' => {
+            'SSL' => false # Disable SSL (by default SAP IGS does not use SSL/TLS)
+          },
+          'DisclosureDate' => '2018-03-14'
+        )
+    )
+    register_options(
+      [
+        Opt::RPORT(40080),
+        OptString.new('FILE', [ false, 'File to read from the remote server', '/etc/passwd']),
+        OptString.new('PATH', [ true, 'Path to the SAP IGS XMLCHART page from the web root', '/XMLCHART']),
```

Comment:
This should really be `URIPATH` to follow the convention of other HTTP modules.

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/14163/commits/41b0bae21fb117ecec9cd3d5330e238ad71c59b3